test-suite/src/test/scala/org/scalajs/testsuite/javalib/ConcurrentHashMapTest.scala

```diff
@@ -0,0 +1,231 @@
+      expect(iter.hasNext).toBeFalsy
+    }
+
+    it("should give proper EntrySet over key values pairs") {
+      val chm = new ConcurrentHashMap[String, String]()
+
+      chm.put("ONE", "one")
+      val entrySet = chm.entrySet
+
+      expect(entrySet.size).toEqual(1)
+
+      val iter = entrySet.iterator
+      expect(iter.hasNext).toBeTruthy
+      val next = iter.next
+      expect(iter.hasNext).toBeFalsy
+      expect(next.getKey).toEqual("ONE")
+      expect(next.getValue).toEqual("one")
+    }
+
+    it("should give proper KeySet over keys") {
+      val chm = new ConcurrentHashMap[String, String]()
+
+      chm.put("ONE", "one")
+      val keySet = chm.keySet
+
+      expect(keySet.size).toEqual(1)
+
+      val iter = keySet.iterator
+      expect(iter.hasNext).toBeTruthy
+      expect(iter.next).toEqual("ONE")
+      expect(iter.hasNext).toBeFalsy
+    }
+
+    it("should put a whole map into") {
+      val chm = new ConcurrentHashMap[String, String]()
+
+      val m = mutable.Map[String, String](
+          "X" -> "y")
+      chm.putAll(mutableMapAsJavaMap(m))
+      expect(chm.size).toEqual(1)
+      expect(chm.get("X")).toEqual("y")
+
+      val nullMap = mutable.Map[String, String](
+          (null , "y"),
+          "X" -> "y")
+
+      expectNullPointerException(chm.putAll(mutableMapAsJavaMap(nullMap)))
+    }
+
+    it("should replace contained items") {
+      val chm = new ConcurrentHashMap[String, String]()
+
+      chm.put("ONE", "one")
+      expect(chm.replace("ONE", "two")).toEqual("one")
+      expectNullPointerException(chm.replace("ONE", null))
+      expectNullPointerException(chm.replace(null, "one"))
+      expect(chm.get("ONE")).toEqual("two")
+
+      expect(chm.replace("ONE", "one", "two")).toBeFalsy
+      expectNullPointerException(chm.replace(null, "two", "one"))
+      expectNullPointerException(chm.replace("ONE", null, "one"))
+      expectNullPointerException(chm.replace("ONE", "two", null))
+
+      expect(chm.replace("ONE", "two", "one")).toBeTruthy
+      expect(chm.get("ONE")).toEqual("one")
+    }
+
+  }
+
+  describe("java.util.AbstractMap.values") {
+
+    case class SimpleQuerableMap[K, V](inner: mutable.HashMap[K, V]) extends java.util.AbstractMap[K, V] {
+      def entrySet(): java.util.Set[java.util.Map.Entry[K, V]] = {
+        setAsJavaSet(
+          inner.map { case (k, v) => new java.util.AbstractMap.SimpleImmutableEntry(k, v) }.toSet)
+      }
+    }    
+
+    it("should check values size") {
+      val chm = new ConcurrentHashMap[String, String]()
+
+      chm.put("ONE", "one")
+      chm.put("TWO", "two")
+
+      expect(chm.values.size).toEqual(2)
+
+      val hm1 = mutable.HashMap(
+        "ONE" -> "one",
+        "TWO" -> "two")
+      val hm2 = mutable.HashMap(
+        "ONE" -> null,
+        "TWO" -> "two")
+      val hm3 = mutable.HashMap(
+        (null, "one"),
+        "TWO" -> "two")
+      val hm4 = mutable.HashMap(
+        (null, null),
+        "TWO" -> "two")
```

Comment:
Sure, don't worry ;-) I'm going to test normal behaviour and corner cases on,
I just ask if you agree to do the tests in the same JasmineTest as ConcurrentHashMapTest within another "describe", and moreover if you are ok with the mocked version to test null corner cases that I cannot test in chm.